infrastructure/lib/constructs/api-sqs-function.ts

```diff
@@ -0,0 +1,65 @@
+import * as iam from "@aws-cdk/aws-iam";
+import * as cdk from "@aws-cdk/core";
+import { HttpMethod } from "../http";
+import { ApiDynamoDBFunction, ApiDynamoDBFunctionProps } from "./api-dynamodb-function";
+import * as sqs from "@aws-cdk/aws-sqs";
+import { SqsEventSource } from "@aws-cdk/aws-lambda-event-sources";
+
+export interface ApiSQSFunctionProps extends ApiDynamoDBFunctionProps {
+  /**
+   * The SQS queue where data is sent
+   */
+  readonly queue: sqs.IQueue;
+  /**
+   * Optional param to create an SQS event source to recieve queue data
+   */
+  readonly createEventSource?: any;
+}
+
+/**
+ * Creates the required resources for a PortfolioDraft function for the ATAT API.
+ *
+ * The route is added to the API Gateway resource and the Lambda function is packaged and
+ * created. Additionally, the necessary permissions are granted on the DynamoDB table
+ * corresponding to the HTTP method specified in the props.
+ */
+export class ApiSQSFunction extends ApiDynamoDBFunction {
+  /**
```

Comment:
I'd love to see a more extensible approach here such as applying the Decorator pattern such that a Function could simply apply whichever Decorators it needed (DynamoDB, SQS, S3, etc.) in lieu of having one extend another. That said, I can see this request sending you down a rabbit hole for at least a day. If this wasn't the last Sprint of the Milestone period, then I'd say give it a shot but let's just leave a big fat TODO to refactor this (perhaps in early M3 depending on whether the Orchestrator is mostly Step Function based or just uses SQS Queues).

Reply:
Also, assuming that we stick with this for now, we still need a UT to test the cases in the `switch` statement.

Reply:
Added to 'ATAT Tech Debt' confluence page

Reply:
yeah... we need a larger strategy around testing the infrastructure code. if there's a good way to test just this function, awesome. otherwise, i think tech debt.

Reply:
We should have Tech Debt for writing unit tests for the constructs anyways, there is no unit test for any construct right now.